### ISSUE TITLE
Editor-Frontend Create New Issue bug

### ISSUE THREAD

**Issue by Enoch-Hsiao** [david-fisher/320-S21-Track1#23]
Code resides in [Editor.js](https://github.com/david-fisher/320-S21-Track1/blob/main/Ghost_in_the_Shell/client_folder_react_app/ethisim/src/pages/editor.js)

**What's supposed to happen:**

When I create a new page, I should be redirected to the newly created page.
![Good](https://user-images.githubusercontent.com/51762139/109838778-7770bf80-7c14-11eb-8f74-020bea910a7a.gif)

**The bug:**

Sometimes, when I create a new page, I **do not** get redirected to the newly created page.
![Bug2](https://user-images.githubusercontent.com/51762139/109838783-793a8300-7c14-11eb-8c99-3a8fcf872024.gif)

As you can see, I get redirected to page "1" (an old page) rather than page "2", the page I just created.

Why does this happen?

In editor.js, the next page index is kept by the variable "addNewPageIndex."
`const [addNewPageIndex, setAddNewPageIndex] = useState(null); \\line193`

Let's walk through the flow of creating a new page:

1. I create a new scenario page by selecting "Add Scenario Page"
2. We execute the _addNewPage_ function:
` const addNewPage = (pageType, pageName, pageBody) => { \\line 495`
3. If we successfully created a new page into the database (POST request), in the _onSuccess_ function, we set addNewPageIndex to scenarioComponents.length.  **_This is the error, because we assume that the new page will be at the end of the array._** Obviously, sometimes the backend does not return the components in order by date created.
```   
\\lines 504-507
            function onSuccess(resp) {
                setAddNewPageIndex(scenarioComponents.length); //Error
                setShouldFetch(shouldFetch + 1);
            }
```
4. Therefore, when we fetch (reload) the updated pages (with the newly created pages), we get redirected to the wrong page.
`let handleLogisticsGet = function handleLogisticsGet() { \\line 196`
```
\\lines 296-276
 if (addNewPageIndex) {
                handlePageGet(
                    setGetValues,
                    initialComponents[addNewPageIndex].id, //Not necessarily the page id I want to fetch!
                    initialComponents
                );
                setAddNewPageIndex(null);
            } 
```
Basically, I fetch with the id of the page at then end of the array (because addNewPageIndex is the last page in the array of pages), but the newly created page is not necessarily the last page in the array.

P.S. Editor.js is a the most disgusting file you'll probably ever see, sorry! Please feel free to leave comments or DM me if you don't understand the logic, need help.

**Comment by AfricanDaniel** [comment]
After a thorough investigation, It seems that the back-end needs to provide additional information (field/variable) to the front-end to indicate which ID is the newly created page corresponding to in the array. 

As mentioned in the above comment, the way it is currently being done in the front end is to pick the last element in the array through scenarioComponents.length. 

We have to rewrite this function so that rather than getting the index through scenarioComponents.length, we will take the new field/variable that should indicate its ID and pass it over to setAddNewPageIndex() to redirect to the correct page.

@aamodpant15 I would highly suggest adding a back-end member to work on this function as well since the data provided to the front-end is very limited and cannot be proceeded unless the back-end functions are taken care of.

**Comment by Enoch-Hsiao** [comment]
> 
> We have to rewrite this function so that rather than getting the index through scenarioComponents.length, we will take the new field/variable that should indicate its ID and pass it over to setAddNewPageIndex() to redirect to the correct page.
> 



```
\\lines 504-507
            function onSuccess(resp) {
                setAddNewPageIndex(scenarioComponents.length); //Error
                setShouldFetch(shouldFetch + 1);
            }
```
When creating a new page (POST req), the backend returns the a response (in the _resp_ variable). It might contain the id for the newly added page.
